lib/Gitlab/Api/Repositories.php

```diff
@@ -4,11 +4,15 @@ class Repositories extends AbstractApi
 {
     /**
      * @param int $project_id
+     * @param array $parameters
      * @return mixed
      */
-    public function branches($project_id)
+    public function branches($project_id, array $parameters = [])
     {
-        return $this->get($this->getProjectPath($project_id, 'repository/branches'));
+        $resolver = $this->createOptionsResolver();
+        $resolver->setDefined('query');
```

Comment:
In api v4 when you try load list of branches you receive Header with pagination. Also pagination query works. 

Note: you can compare queries.
 
V3: 
http://take.ms/e4hJW

V4:
http://take.ms/NNhQK
http://take.ms/8AD67